### PR TITLE
feat: add better type hints for impersonate and session

### PR DESCRIPTION
**Pull request by vvanglro**
Hints for enhancing `impersonate`.

For method impersonate:

image

For session impersonate:

image

For session kwargs:

image

### CODE REVIEW

File: curl_cffi/requests/impersonate.py [hunk 1]
```diff
@@ -1,49 +1,59 @@
...
 from dataclasses import dataclass
 from typing import List, Literal, Optional, TypedDict
-import warnings
-from enum import Enum
-
-from ..const import CurlSslVersion, CurlOpt
 
+from ..const import CurlOpt, CurlSslVersion
 
-class BrowserType(str, Enum):
```
[the remark this answers is not in the document]
Comment (vvanglro) [reply curl_cffi/requests/impersonate.py:9]: Makes sense.

[the remark this answers is not in the document]
Comment (yifeikong) [reply curl_cffi/requests/impersonate.py:9]: I agree. I'm open to remove this class in major version bump, e.g. `1.x`, but let's try to keep it for now.

File: curl_cffi/requests/impersonate.py [hunk 2]
```diff
@@ -1,12 +1,63 @@
...
+    "safari_ios",
+    "chrome_android",
+]
 
-class BrowserType(str, Enum):
+DefaultChrome = "chrome124"
+DefaultEdge = "edge101"
+DefaultSafari = "safari17_0"
+DefaultSafariPhone = "safari17_2_ios"
+DefaultChromePhone = "chrome99_android"
```
Comment (yifeikong) [inline curl_cffi/requests/impersonate.py:42]: For constants, `DEFAULT_CHROME` is preferred over `DefaultChrome`.

File: curl_cffi/requests/impersonate.py [hunk 3]
```diff
@@ -1,12 +1,63 @@
...
+
+
+def normalize_browser_type(item):
+    if item == "chrome":  # noqa: SIM116
+        return DefaultChrome
```
Comment (yifeikong) [inline curl_cffi/requests/impersonate.py:47]: Same here.

File: curl_cffi/requests/session.py [hunk 4]
```diff
@@ -24,20 +24,22 @@
...
 from .impersonate import (
+    TLS_CIPHER_NAME_MAP,
+    TLS_EC_CURVES_MAP,
+    TLS_VERSION_MAP,
+    BrowserTypeLiteral,
     ExtraFingerprints,
     ExtraFpDict,
+    normalize_browser_type,
     toggle_extension,
-    BrowserType,
```
Comment (yifeikong) [inline curl_cffi/requests/session.py:37]: This line should be kept for backward compatibility.

File: curl_cffi/requests/impersonate.py (hunk 2 above)
Comment (yifeikong) [reply curl_cffi/requests/impersonate.py:42]: `DefaultChromePhone` should be `DEFAULT_CHROME_ANDROID`. All browsers on iOS uses the webkit core, which means they use the webkit fingerprints.

**Pull request merged**